src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMProbe.java

```diff
@@ -151,20 +153,39 @@ public SCMProbeStat stat(@NonNull String path) throws IOException {
                     return SCMProbeStat.fromAlternativePath(content.getPath());
                 }
             }
-        } catch (FileNotFoundException fnf) {
+        } catch (GHFileNotFoundException fnf) {
+            boolean finicky = false;
             if (index == 0 || index == 1) {
-                // the revision does not exist, we should complain.
-                throw fnf;
+                // the revision does not exist, we should complain unless JENKINS-54126.
+                finicky = true;
             } else {
                 try {
                     repo.getDirectoryContent("/", Constants.R_REFS + ref);
                 } catch (IOException e) {
-                    // this must be an issue with the revision, so complain
+                    // this must be an issue with the revision, so complain unless JENKINS-54126
                     fnf.addSuppressed(e);
-                    throw fnf;
+                    finicky = true;
                 }
                 // means that does not exist and this is handled below this try/catch block.
             }
+            if (finicky) {
+                final Optional<List<String>> status = Optional.ofNullable(fnf.getResponseHeaderFields() != null ? fnf.getResponseHeaderFields().get(null) : null);
+                if (GitHubSCMSource.getCacheSize() > 0
+                        && gitHub.getConnector() instanceof Connector.ForceValidationOkHttpConnector
+                        && status.isPresent() && status.get().stream().anyMatch((s) -> s.contains("40"))) { //Any status >= 400 is a FNF in okhttp
+                    //JENKINS-54126 try again without cache headers
+                    final Connector.ForceValidationOkHttpConnector oldConnector = (Connector.ForceValidationOkHttpConnector) gitHub.getConnector();
+                    try {
+                        //TODO I'm not sure we are alone in using this connector so maybe concurrent modification problems
```

Comment:
maybe have a withoutCache method that takes a closure, that way you could use a lock in the connector to ensure the correct one gets used... if you are worried about concurrent requests